src/scripts/main.js

```diff
@@ -1,3 +1,67 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const appendRowBtn = document.querySelector('.append-row');
+const removeRowBtn = document.querySelector('.remove-row');
+const appendColumnBtn = document.querySelector('.append-column');
+const removeColumnBtn = document.querySelector('.remove-column');
+
+const maxRows = 10;
+const minRows = 2;
+const maxCols = 10;
+const minCols = 2;
+
+const updateButtonsState = () => {
+  const rowCount = table.rows.length;
+  const colCount = table.rows[0] ? table.rows[0].cells.length : 0;
```

Comment:
The code assumes that there is always at least one row in the table when checking the column count. If the table is empty, this could lead to an error. Consider adding a check to ensure the table has rows before accessing `table.rows[0]`.